tests/lax_numpy_test.py

```diff
@@ -286,6 +286,59 @@ def np_fun(x):
                             atol={dtypes.bfloat16: 1e-1, np.float16: 1e-2})
     self._CompileAndCheck(jnp_fun, args_maker, atol={dtypes.bfloat16: 1e-1})
 
+
+  @jtu.sample_product(
+    [dict(shape=shape, axis=axis)
+      for shape in all_shapes
+      for axis in list(range(-len(shape), len(shape))) + [None] if len(shape) == 1],
+    dtype=all_dtypes,
+    out_dtype=all_dtypes + [None],
+    include_initial=[False, True],
+  )
+  @jax.numpy_dtype_promotion('standard')  # This test explicitly exercises mixed type promotion
+  def testCumulativeSum(self, shape, axis, dtype, out_dtype, include_initial):
+    rng = jtu.rand_some_zero(self.rng())
+    x = rng(shape, dtype)
+    out = jnp.cumulative_sum(x, dtype=out_dtype, include_initial=include_initial)
+
+    target_dtype = out_dtype or x.dtype
+    kind = x.dtype.kind
+    if (out_dtype is None and kind in {'i', 'u'}
+        and x.dtype.itemsize*8 < int(config.default_dtype_bits.value)):
+      target_dtype = dtypes.dtype(dtypes._default_types[kind])
+    assert out.dtype == target_dtype
+
+    _axis = axis or 0
+    target_shape = list(x.shape)
+    if include_initial:
+      target_shape[_axis] += 1
+    assert out.shape == tuple(target_shape)
+
+    target = jnp.cumsum(x, axis=_axis, dtype=out.dtype)
+    if include_initial:
+      zeros_shape = target_shape
+      zeros_shape[_axis] = 1
+      target = jnp.concat([jnp.zeros(target_shape, dtype=out.dtype), target])
+    self.assertArraysEqual(out, target)
```

Comment:
Please also use `self._CompileAndCheck`, which asserts that the function's semantics don't change under JIT.